### ISSUE TITLE
Max CPU load with any changes

### ISSUE THREAD

**Issue by MrHHatter** [Beep6581/RawTherapee#5191]
**Version: 5.5**
Branch: 5.5
Commit: d9ac23555
Commit date: 2018-12-17
Compiler: gcc 8.2.1
Processor: generic x86
System: Windows
Bit depth: 64 bits
Gtkmm: V3.22.3
Lensfun: V0.3.2.0
Build type: release
Build flags:  -m64 -mwin32 -msse2 -mfpmath=sse -mthreads -Wno-aggressive-loop-optimizations -Wno-parentheses -std=c++11 -mtune=generic -Werror=unused-label -Wall -Wuninitialized -Wno-deprecated-declarations -Wno-unused-result -fopenmp -Werror=unknown-pragmas  -DNDEBUG  -O3 -ftree-vectorize
Link flags: -m64  -mthreads  -static-libgcc   -mtune=generic  -s  -O3  -fno-use-linker-plugin
OpenMP support: ON
MMAP support: ON

**Operating system Windows 10 Home 64-bit**
AMD Ryzen 5 2600X
16GB G.Skill DDR4 Ram at 3000MHz

**Issue:**
When editing photos, whenever I change any parameters, CPU goes to max load on all cores and it takes a few seconds for the changes to take place. There is no fluidity when making changes. Makes no difference what type of changes is applied. Even zooming has the same effect on CPU load. It's an always present issue.
Editing Pentax .dng-files with an average size of 15mb.

**Comment by heckflosse** [comment]
@MrHHatter Please provide a .pp3 file

**Comment by MrHHatter** [comment]
@heckflosse Recent full edit .pp3:
https://filebin.net/eo5alhzwxkf2q8ii

**Comment by heckflosse** [comment]
@MrHHatter It's mainly caused by Noise reduction. To verify: zooming at less than 100% should be fast.

**Comment by MrHHatter** [comment]
@heckflosse Nope. The same happens without Noise reduction. On this one I did nothing but open the image and raise the exposure. When changing the exposure, RT slows down, CPU usage goes to max on all cores and the changes applies after a few seconds.
https://filebin.net/htv96ms3vqhd87xp

**Comment by Thanatomanic** [comment]
Just to check: have you stored the file on a HDD, SSD or on a network drive?

**Comment by MrHHatter** [comment]
@Thanatomanic It's stored on a Samsung M2 SSD 960 EVO. At first I stored the files on a high speed HDD but trying to fix the issue I moved the images I want to edit to the M2 drive. Made no difference at all.

**Comment by heckflosse** [comment]
@MrHHatter What's your screen resolution?

**Comment by MrHHatter** [comment]
@heckflosse 3840x2160

**Comment by heckflosse** [comment]
@MrHHatter Are you using rawtherapee-debug.exe by accident?

**Comment by MrHHatter** [comment]
@heckflosse Nope, dubble checked. I am not.

**Comment by heckflosse** [comment]
@MrHHatter Can you provide a raw file for the .pp3 from above?

**Comment by heckflosse** [comment]
@MrHHatter And a screenshot showing ratherapee with an image opened in editor, please.

**Comment by MrHHatter** [comment]
![printscreen](https://user-images.githubusercontent.com/47943904/53301147-6f286180-384f-11e9-8f3b-68611159f20a.jpg)

https://filebin.net/q1skbnvysby1o241 to the RAW file

**Comment by Thanatomanic** [comment]
@MrHHatter @heckflosse My edits to this file are nearly instantaneous... I can make a screencast if proof is needed.

**Comment by MrHHatter** [comment]
Never had this problem in either Lightroom or PhotoScape X, even on my old, much slower computer. The system or the files are not the cause of the problem.

**Comment by heckflosse** [comment]
@MrHHatter @Thanatomanic 
In RT 5.5. release we set Lens-profile Chromatic abberation correction enabled in the Auto-matched profiles. That causes a slight slowdown and also does not work well together with the also enabled raw ca-correction.

@MrHHatter Can you please try to disable chromatic abberation correction in profiled lens correction.
Here it gets a bit faster by disabling it (though it's already fast before, but I'm not using such a large resolution).

![grafik](https://user-images.githubusercontent.com/13733487/53301373-61281000-3852-11e9-8f1c-92d2dc98fceb.png)

**Comment by heckflosse** [comment]
@Thanatomanic Will your screencast use the official 5.5 or your own build?

**Comment by heckflosse** [comment]
@MrHHatter 
> even on my old, much slower computer

Was that with same screen size? Processing time for RT editor increases with resolution alomst linearly.

**Comment by MrHHatter** [comment]
@heckflosse the Chromatic aberration possibly made a slight difference but I can't really tell. I did however try to lower my resolution to 1920x1080 and that made a difference. It's still far from fluid but the changes takes maybe half a second instead of several seconds.
I've not used Lightroom with the 4k resolution but I tried PhotoScape X just moments ago, at 4K, and it had no problem what so ever to handle it. Given, it's not as powerful as RT, but the CPU goes up to about 30% at most, no matter what PhotoScape X throws at it.

**Comment by heckflosse** [comment]
There's definitely a difference in speed between official build and my own native build (optimized for my AMD FX 8350)

**Comment by Thanatomanic** [comment]
@heckflosse I've uploaded my screencast in this filebin: https://filebin.net/q1skbnvysby1o241
I use a fresh install of 5.5 on a 1920x1080 screen. And yes, I also notice a minor but apparent speed difference between my own build and the release version.

**Comment by heckflosse** [comment]
@Thanatomanic Your screencast shows about the same speed as using a fresh 5.5 install on my machine.
